packages/zoe/src/contracts/priceAggregatorChainlink.js

```diff
@@ -557,122 +656,170 @@ const start = async (
-   *   oracleRoundState(_oracle: OracleKey, _queriedRoundId: BigInt): Promise<any>
+   * @type {Omit<import('./priceAggregator').PriceAggregatorContract['creatorFacet'], 'initOracle'> & {
+   *   initOracle: (instance) => Promise<OracleAdmin<PriceRound>>,
+   *   getRoundData(roundId: bigint | number): Promise<RoundData>,
+   *   oracleRoundState(oracleAddr: string, queriedRoundId: BigInt): Promise<any>
    * }}
    */
   const creatorFacet = Far('PriceAggregatorChainlinkCreatorFacet', {
-    deleteOracle: async oracleKey => {
-      const records = keyToRecords.get(oracleKey);
+    /**
+     * An "oracle invitation" is an invitation to be able to submit data to
+     * include in the priceAggregator's results.
+     *
+     * The offer result from this invitation is a OracleAdmin, which can be used
+     * directly to manage the price submissions as well as to terminate the
+     * relationship.
+     *
+     * @param {string} oracleAddr
+     */
+    makeOracleInvitation: async oracleAddr => {
+      /**
+       * If custom arguments are supplied to the `zoe.offer` call, they can
+       * indicate an OraclePriceSubmission notifier and a corresponding
+       * `shiftValueOut` that should be adapted as part of the priceAuthority's
+       * reported data.
+       *
+       * @param {ZCFSeat} seat
+       * @returns {Promise<{admin: OracleAdmin<PriceRound>, invitationMakers: {makePushPriceInvitation: (result: PriceRound) => Promise<Invitation<void>>} }>}
+       */
+      const offerHandler = async seat => {
+        const admin = await creatorFacet.initOracle(oracleAddr);
+        const invitationMakers = Far('invitation makers', {
+          /** @param {PriceRound} result */
+          makePushPriceInvitation(result) {
+            assertParsableNumber(result.unitPrice);
+            return zcf.makeInvitation(cSeat => {
+              cSeat.exit();
+              admin.pushResult(result);
+            }, 'PushPrice');
+          },
+        });
+        seat.exit();
+
+        return harden({
+          admin,
+
+          invitationMakers,
+        });
+      };
+
+      return zcf.makeInvitation(offerHandler, INVITATION_MAKERS_DESC);
+    },
+    /** @param {string} oracleAddr */
+    deleteOracle: async oracleAddr => {
+      const records = keyToRecords.get(oracleAddr);
       for (const record of records) {
         records.delete(record);
       }
 
-      oracleStatuses.delete(oracleKey);
+      oracleStatuses.delete(oracleAddr);
 
       // We should remove the entry entirely, as it is empty.
-      keyToRecords.delete(oracleKey);
+      keyToRecords.delete(oracleAddr);
     },
 
     // unlike the median case, no query argument is passed, since polling behavior is undesired
-    async initOracle(oracleInstance) {
-      /** @type {OracleKey} */
-      const oracleKey = oracleInstance || Far('oracleKey', {});
-
+    /** @param {string} oracleAddr */
+    async initOracle(oracleAddr) {
```

Comment:
Is this really a string? The tests suggest otherwise:

```js
  const pricePushAdminA = await E(aggregator.creatorFacet).initOracle(
    priceOracleA.instance,
  );
```

Reply:
Thanks for raising. That test was written for the agsolo era. We only support smart wallet oracles now so I've amended to make that clear. This is also necessary for durability.